### PR TITLE
Web3 ID logging

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Add logging of failure events for easier debugging.

## Changes
- Added logging entry to `chrome.storage`. It will currently store the most recent 100 events.
- For the development build errors are also logged to `console`.
- Added logging to the Web3 ID error scenarios I could find where we wanted it to be logged.

## Checklist
- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: packages/browser-wallet/src/shared/storage/update.ts [hunk 1]
```diff
@@ -60,6 +60,28 @@ export async function removeFromList<Type>(
     });
 }
 
+/*
+ * Generic method to add an element to list in storage while ensuring that
+ * the list never grows beyond the provided size. If the list is still small
+ * enough, then the addition is prepended to the list. If the the list would have
+ * grown greater than the max size, then the addition is prepended to the list and
+ * the last element of the list is removed.
+ */
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/shared/storage/update.ts:69]:
```suggestion
/*
 * Generic method to add an element to list in storage, while treating the list as a FIFO queue, to ensure the list never grows beyond the provided size.
 */
``````

File: packages/browser-wallet/src/shared/utils/log-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { storedLog } from '@shared/storage/access';
+import { addToListMaxSize } from '@shared/storage/update';
+import { isDevelopmentBuild } from './environment-helpers';
+
+const loggingLock = 'concordium_log_lock';
+
+// Determines the maximum number of log entries that is stored
+// at a time.
+const LOG_MAX_ENTRIES = 100;
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/shared/utils/log-helpers.ts:9]: Maybe define this in constants.

File: packages/browser-wallet/src/shared/utils/log-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+function logForDevelopmentBuild(logEntry: Log) {
+    const logMessage = `[${new Date(logEntry.timestamp).toISOString()}] ${logEntry.level} ${logEntry.message}`;
+    switch (logEntry.level) {
+        case LoggingLevel.WARN:
+            // eslint-disable-next-line no-console
+            console.warn(logMessage);
+            break;
+        case LoggingLevel.ERROR:
+            // eslint-disable-next-line no-console
+            console.error(logMessage);
```
Comment (shjortConcordium) [inline packages/browser-wallet/src/shared/utils/log-helpers.ts:36]: Could be nice to have stack trace here if its an error.

File: packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialHooks.tsx [hunk 4]
```diff
@@ -199,10 +205,12 @@ export function useIssuerMetadata(issuer: string): IssuerMetadata | undefined {
...
-        getCredentialRegistryMetadata(client, registryContractAddress).then((res) => {
-            const abortController = new AbortController();
-            fetchIssuerMetadata(res.issuerMetadata, abortController).then(setIssuerMetadata);
-        });
+        getCredentialRegistryMetadata(client, registryContractAddress)
+            .then((res) => {
+                const abortController = new AbortController();
+                fetchIssuerMetadata(res.issuerMetadata, abortController).then(setIssuerMetadata).catch(logError);
+            })
+            .catch(logError);
```
[the remark this answers is not in the document]
Comment (orhoj) [reply packages/browser-wallet/src/popup/pages/VerifiableCredential/VerifiableCredentialHooks.tsx:213]: It is a bug that it isn't used here. So I am now using it to abort in the cleanup.

File: packages/browser-wallet/src/shared/utils/log-helpers.ts (hunk 2 above)
Comment (orhoj) [reply packages/browser-wallet/src/shared/utils/log-helpers.ts:9]: Good idea.

File: packages/browser-wallet/src/shared/utils/log-helpers.ts (hunk 3 above)
Comment (orhoj) [reply packages/browser-wallet/src/shared/utils/log-helpers.ts:36]: Let's do that soonTM

**Pull request merged**